codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordEncoder.java

```diff
@@ -101,4 +176,13 @@ protected void encodeName(String name, ByteBuf buf) throws Exception {
 
         buf.writeByte(0); // marks end of name field
     }
+
+    // Package private so it can be reused in the test.
+    static byte padWithZeros(byte b, int leftOverBits) {
+        int lastBit = Byte.SIZE - leftOverBits;
```

Comment:
actually it looks like `leftOverBits` is used more like `bitsToKeep` and so the case statements should be reversed.